addons/mail/models/ir_mail_server.py

```diff
@@ -1,18 +1,23 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import email_normalize
 
 
 class IrMail_Server(models.Model):
     _inherit = 'ir.mail_server'
+    _email_field = 'smtp_user'
 
     mail_template_ids = fields.One2many(
         comodel_name='mail.template',
         inverse_name='mail_server_id',
         string='Mail template using this mail server',
         readonly=True)
 
+    owner_id = fields.Many2one('res.users', 'Owner')
```

Comment:
Should we have a UNIQUE constraint on that? To avoid a single user creating multiple personal external mail servers?